osu.Game/Users/User.cs

```diff
@@ -69,6 +69,9 @@ public class UserCover
         [JsonProperty(@"support_level")]
         public int SupportLevel;
 
+        [JsonProperty(@"current_mode_rank")]
+        public int? CurrentModeRank;
```

Comment:
In my opinion this is even worse than `APIFriend` was - now the base user has a property that only one overlay will use.

I would bring back `APIFriend`, and actually *duplicate* the `User` fields that are returned by the API, and *only* those (so as far as I can see from web: `'cover', 'country', 'current_mode_rank', 'support_level'`, and then everything else from [`UserCompact`](https://osu.ppy.sh/docs/index.html#usercompact)). Heck, you could even make an `APIUserCompact` and then derive from *that*, adding the other fields.

I wouldn't be scared of specialised data structures for API interfacing. The concept of data transfer objects has been alive for long.

Reply:
IMO it's fine for now. We already have properties in `User` which have been used only in one particular case. I agree, that this should be fixed at some point, but at very least it deserves a separate pr (which I'm not gonna make since I'm not sure how to do that properly)

Reply:
I am unconvinced but will defer on this if it turns out that it is only my opinion that this should be fixed here.

Reply:
Or probably I should make `APIFriend : APIUser` and place that property there

Reply:
In RL discussions, we were talking about moving all API pieces of the user to `APIUser` and derive from that. Which is similar to what @bdach is suggesting.

ie. all serialization logic should be moved out of User and there should be a `ToUser()` function on `APIUser` / whatever, simlar to `APIBeatmapSet`'s implementation.

To clarify, there's no problem with deriving/subclassing if it's API specific models.